lib/SConscript

```diff
@@ -1,7 +1,15 @@
 # -*- python -*-
 from lsst.sconsUtils import scripts, targets, env
+import os
 
-for flag in ("-fexceptions", "-DNSUPERNODAL", "-DNPARTITION"):
+
+flags = ["-fexceptions", "-DNSUPERNODAL", "-DNPARTITION"]
+
+uname = os.uname()
+if uname.sysname == 'Darwin' and uname.machine == 'arm64':
+    flags.append('-faligned-allocation')
```

Comment:
Is there a down side to using that on intel macOS as well?

Reply:
If it ain't broke...
But I don't know.

Reply:
Looks like it's only available for C++17 compilers.

```
-faligned-allocation    Enable C++17 aligned allocation functions
```
We are meant to be using C++17 these days but this is presumably fall out from us insisting on using an old macOS deployment target on Intel.

Reply:
Yes, a little more research (and as noted in the ticket description) I think that the deployment target that we have as a default on x86 will make this impossible to have as the universal default.

Reply:
Enabling C++17 compilation doesn't necessarily enable library support for it. MacOSx way to be backwards
compatible with old deployment targets.